sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -126,9 +126,13 @@ public virtual async Task ReceiveAndProcessMessagesAsync(CancellationToken cance
 
         public virtual void UpdatePrefetchCount(int prefetchCount)
         {
-            if (Receiver != null && Receiver.PrefetchCount != prefetchCount)
+            var capturedReceiver = Receiver;
+
+            // If the Receiver property is set to null after we have captured a non-null instance, the Prefetch setter will essentially
+            // no-op as the underlying link has been closed.
+            if (capturedReceiver != null && capturedReceiver.PrefetchCount != prefetchCount)
```

Comment:
If the captured receiver is not the current `Receiver` and we change the prefetch, what happens?   Is this a case where we potentially think we've updated the value but the new `Receiver` isn't aware?

Should we consider doing a reference check and resetting captured receiver if we don't have a `ReferenceEquals` match between `capturedReceiver` and `Receiver`?

Reply:
> If the captured receiver is not the current `Receiver` and we change the prefetch, what happens? Is this a case where we potentially think we've updated the value but the new `Receiver` isn't aware?

It's possible, but benign. The prefetch count would be updated on the next iteration of the RunReceiveTask loop. There isn't a way to eliminate this possibility without locking.

> Should we consider doing a reference check and resetting captured receiver if we don't have a ReferenceEquals match between capturedReceiver and Receiver?

The only way the Receiver would be different from the captured receiver is if the captured receiver has now been closed. In this case, the PrefetchCount update is a no-op as the link is closed.